staker/rollup_watcher.go

```diff
@@ -180,17 +181,28 @@ func (r *RollupWatcher) LookupNodeChildren(ctx context.Context, nodeNum uint64,
 		Addresses: []common.Address{r.address},
 		Topics:    [][]common.Hash{{nodeCreatedID}, nil, {nodeHash}},
 	}
-	query.FromBlock, err = r.getNodeCreationBlock(ctx, nodeNum)
+	fromBlock, err := r.getNodeCreationBlock(ctx, nodeNum)
 	if err != nil {
 		return nil, err
 	}
-	query.ToBlock, err = r.getNodeCreationBlock(ctx, node.LatestChildNumber)
+	toBlock, err := r.getNodeCreationBlock(ctx, node.LatestChildNumber)
 	if err != nil {
 		return nil, err
 	}
-	logs, err := r.client.FilterLogs(ctx, query)
-	if err != nil {
-		return nil, err
+	var logs []types.Log
+	// break down the query to avoid eth_getLogs query limit
+	for toBlock.Cmp(fromBlock) > 0 {
+		query.FromBlock = fromBlock
+		query.ToBlock = new(big.Int).Add(fromBlock, big.NewInt(int64(logQueryRangeSize)))
```

Comment:
You'll need to change the logic so that all blocks are requested when logQueryRangeSize is zero